shell/platform/android/io/flutter/embedding/engine/systemchannels/AccessibilityChannel.java

```
@@ -0,0 +1,91 @@
+      // parse it. Return.
+      if (handler == null) {
+        return;
+      }
+
+      @SuppressWarnings("unchecked")
+      final HashMap<String, Object> annotatedEvent = (HashMap<String, Object>) message;
+      final String type = (String) annotatedEvent.get("type");
+      @SuppressWarnings("unchecked")
+      final HashMap<String, Object> data = (HashMap<String, Object>) annotatedEvent.get("data");
+
+      switch (type) {
+        case "announce":
+          handler.announce((String) data.get("message"));
+          break;
+        case "tap": {
+          Integer nodeId = (Integer) annotatedEvent.get("nodeId");
+          if (nodeId != null) {
+            handler.onTap(nodeId);
+          }
+          break;
+        }
+        case "longPress": {
+          Integer nodeId = (Integer) annotatedEvent.get("nodeId");
+          if (nodeId != null) {
+            handler.onLongPress(nodeId);
+          }
+          break;
+        }
+        case "tooltip": {
+          handler.tooltip((String) data.get("message"));
+          break;
+        }
+      }
+    }
+  };
+
+  AccessibilityChannel(@NonNull DartExecutor dartExecutor) {
+    channel = new BasicMessageChannel<>(dartExecutor, "flutter/accessibility", StandardMessageCodec.INSTANCE);
+    channel.setMessageHandler(parsingMessageHandler);
+  }
+
+  public void release() {
+    handler = null;
+  }
+
+  public void setAccessibilityMessageHandler(@Nullable AccessibilityMessageHandler handler) {
+    this.handler = handler;
+  }
+
+  // TODO(mattcarroll): javadoc
+  public void overrideDefaultMethodHandler(@NonNull BasicMessageChannel.MessageHandler<Object> messageHandler) {
+    channel.setMessageHandler(messageHandler);
+  }
+
+  // TODO(mattcarroll): javadoc
+  public void restoreDefaultMethodHandler() {
+    channel.setMessageHandler(parsingMessageHandler);
+  }
+
+  public interface AccessibilityMessageHandler {
+    void announce(@NonNull String message);
```

This and tooltip are annotated as nonNull, but you don't null check above

---

@jonahwilliams what do you think the appropriate action is if the message is null from the other side of the channel?  Should we fizzle and not invoke announce()?  Or should we throw an exception?  I guess the question is if this mistake should kill the app...

---

Resolved offline